app/src/main/java/org/mozilla/fenix/compose/MessageCard.kt

```diff
@@ -0,0 +1,223 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.compose
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Card
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import org.mozilla.experiments.nimbus.StringHolder
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.button.PrimaryButton
+import org.mozilla.fenix.gleanplumb.Message
+import org.mozilla.fenix.nimbus.MessageData
+import org.mozilla.fenix.nimbus.StyleData
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.theme.Theme
+
+/**
+ * Message Card.
+ *
+ * @param message [Message] that holds a representation of GleanPlum message from Nimbus.
+ * @param onClick Invoked when user clicks on the message card.
+ * @param onCloseButtonClick Invoked when user clicks on close button to remove message.
+ */
+@Suppress("LongMethod")
+@Composable
+fun MessageCard(
+    message: Message,
+    onClick: () -> Unit,
+    onCloseButtonClick: () -> Unit,
+) {
+    Card(
+        modifier = Modifier
+            .padding(
+                top = 16.dp,
+                bottom = 16.dp
+            )
+            .then(
+                if (message.data.buttonLabel.isNullOrBlank()) {
+                    Modifier.clickable(onClick = onClick)
+                } else {
+                    Modifier
+                }
+            ),
+        shape = RoundedCornerShape(8.dp),
+        backgroundColor = FirefoxTheme.colors.layer2,
+    ) {
+        Column(
+            Modifier
+                .padding(all = 16.dp)
+                .fillMaxWidth()
+        ) {
+            val title = message.data.title
+            if (!title.isNullOrBlank()) {
+                Row(
+                    modifier = Modifier.fillMaxWidth(),
+                ) {
+                    SectionHeader(
+                        text = title,
+                        modifier = Modifier
+                            .weight(1f)
+                    )
```

Comment:
Can you move the if statement within the `Row`? Currently, the Row and IconButton is common between both cases, and you can add an if statement check for the PrimaryText in line 99.

Reply:
No, because if the title is not null, `SectionHeader` and `IconButton` needs to be in the `Row` while `PrimaryText` is out but in the else case, `PrimaryText` comes within the `Row`, hence the if statement surrounds the `Row`